specification/iotcentral/data-plane/Microsoft.IoTCentral/stable/1.0.0/iotcentral.json

```diff
@@ -0,0 +1,2262 @@
+            "schema": {
+              "$ref": "#/definitions/Device"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Devices_Create",
+        "summary": "Create or update a device",
+        "description": "Create a new device.",
+        "x-ms-examples": {
+          "Create device": {
+            "$ref": "./examples/devices_create.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device body.",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "Devices_Update",
+        "summary": "Update a device via patch",
+        "description": "Update an existing device by ID.",
+        "x-ms-examples": {
+          "Update device": {
+            "$ref": "./examples/devices_create.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device patch body.",
+            "schema": {
+              "type": "object"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "Devices_Remove",
+        "summary": "Delete a device",
+        "description": "Delete an existing device by device ID.",
+        "x-ms-examples": {
+          "Delete device": {
+            "$ref": "./examples/devices_delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Success"
+          }
+        }
+      }
+    },
+    "/devices/{device_id}/attestation": {
+      "get": {
+        "operationId": "Devices_GetAttestation",
+        "summary": "Get device attestation",
+        "x-ms-examples": {
+          "Get device attestation": {
+            "$ref": "./examples/devices_attestations_get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Attestation"
+            }
+          }
+        }
+      },
+      "put": {
```

Comment:
Seems odd that one can PUT on a REST path that doesn't have a unique Id at the end. One would expect /devices/{device_id}/attestation/{attestationId}. I guess there can only be one attestation.

What is the guidance for PUTs on things like this?

Reply:
@johanste @JeffreyRichter ?

Reply:
yes - unless this is a "singleton" (or at least a well-known name), PUT is suspicious.

Reply:
This is "Singleton", there only 1 attestation per device.